### PR TITLE
p2p(ticdc): remove message peer when capture is removed

### LINKED ISSUE #8922: p2p(ticdc): remove message peer when capture is removed
p2p(ticdc): remove message peer when capture is removed

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

If an owner is killed by "kill -19",  other CDC nodes will remove that capture, but the peer in the message server will not be removed, so the message server still sends ack message to that peer, until the write buffer is full.

Issue Number: close #8922

### What is changed and how it works?
remove message peer when capture is removed

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: pkg/p2p/server.go [hunk 1]
```diff
@@ -287,6 +291,9 @@ func (m *MessageServer) run(ctx context.Context) error {
 					}
 					return errors.Trace(err)
 				}
+			case taskOnDeregisterPeer:
+				log.Debug("taskOnDeregisterPeer", zap.String("peerID", task.peerID))
```
Comment (nongfushanquan) [inline pkg/p2p/server.go:295]: Log.Info level is suggested here

File: cdc/capture/capture.go [hunk 2]
```diff
@@ -459,6 +459,11 @@ func (c *captureImpl) campaignOwner(ctx cdcContext.Context) error {
 		})
 		globalState.SetOnCaptureRemoved(func(captureID model.CaptureID) {
 			c.MessageRouter.RemovePeer(captureID)
+			if err := c.MessageServer.ScheduleDeregisterPeerTask(ctx, captureID); err != nil {
```
Comment (overvenus) [inline cdc/capture/capture.go:462]: Could you add some comments?

**Review by overvenus — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

File: pkg/p2p/server.go (hunk 1 above)
Comment (sdojjy) [reply pkg/p2p/server.go:295]: done

File: cdc/capture/capture.go (hunk 2 above)
Comment (sdojjy) [reply cdc/capture/capture.go:466]: done

**Pull request merged**